docs/csharp/whats-new/tutorials/compound-assignment-operators.md

```diff
@@ -0,0 +1,159 @@
+
+To see this allocation behavior in action, try running the [.NET Object Allocation tracking tool](/visualstudio/profiling/dotnet-alloc-tool) in Visual Studio. When you profile the current implementation during the concert simulation, you discover that it allocates 134 `GateAttendance` objects to complete the relatively small simulation. Each operator call creates a new instance, demonstrating how quickly allocations can accumulate in real-world scenarios. This measurement provides a concrete baseline for comparing the performance improvements you achieve with compound assignment operators.
+
+## Implement compound assignment operators
+
+C# 14 introduces user-defined compound assignment operators that enable in-place mutations instead of creating new instances. These operators provide a more efficient alternative to the traditional pattern while maintaining the familiar compound assignment syntax.
+
+Compound assignment operators use a new syntax that declares `void` return methods with the `operator` keyword. Add the following operators to the `GateAttendance` class:
+
+```csharp
+public void operator +=(int value) => this.property += value;
+public void operator ++() => this.property++;
+```
+
+The key differences from traditional operators are:
+
+- **Mutation**: They modify the current instance directly using `this`
+- **No new instances**: Unlike traditional operators that return new objects, compound operators modify existing ones
+- **Return type**: Compound assignment operators return `void`, not the type itself
+
+When the compiler encounters compound assignment expressions like `a += b` or `++a`, it follows this resolution order:
+
+1. **Check for compound assignment operator**: If the type defines a user-defined compound assignment operator (`+=`, `++`, etc.), use it directly
+2. **Fallback to traditional expansion**: If no compound operator exists, expand to the traditional form (`a = a + b`)
+
+This means you can implement both approaches simultaneously. The compound operators take precedence when available, but the traditional operators serve as fallbacks for scenarios where compound assignment isn't suitable.
+
+Compound assignment operators provide several advantages:
+
+- **Reduced allocations**: Modify objects in-place instead of creating new instances
+- **Improved performance**: Eliminate temporary object creation and reduce garbage collection pressure  
+- **Familiar syntax**: Use the same `+=`, `++` syntax developers already know
+- **Backward compatibility**: Traditional operators continue to work as fallbacks
```

Comment:
```suggestion
- **Reduced allocations**: Modify objects in-place instead of creating new instances.
- **Improved performance**: Eliminate temporary object creation and reduce garbage collection pressure.
- **Familiar syntax**: Use the same `+=`, `++` syntax developers already know.
- **Backward compatibility**: Traditional operators continue to work as fallbacks.
```